docs/local-development.md

```diff
@@ -0,0 +1,226 @@
+# Local Development Setup Guide
+
+This guide will help you set up a local development environment for the Restarters.net project using Docker.
+
+- [Local Development Setup Guide](#local-development-setup-guide)
+  - [Prerequisites](#prerequisites)
+  - [Setup Steps](#setup-steps)
+    - [1. Clone the Repository](#1-clone-the-repository)
+    - [2. Environment Configuration](#2-environment-configuration)
+    - [3. Starting the Development Environment](#3-starting-the-development-environment)
+      - [Using Task (Recommended)](#using-task-recommended)
+      - [Using Docker Compose Directly](#using-docker-compose-directly)
+    - [4. Initial Setup](#4-initial-setup)
+    - [5. Accessing the Application](#5-accessing-the-application)
+  - [Common Tasks](#common-tasks)
+    - [Running Commands in the Container](#running-commands-in-the-container)
+      - [Using Task](#using-task)
+      - [Using Docker Directly](#using-docker-directly)
+    - [Stopping the Environment](#stopping-the-environment)
+      - [Using Task](#using-task-1)
+      - [Using Docker Compose](#using-docker-compose)
+    - [Rebuilding Containers](#rebuilding-containers)
+      - [Using Task](#using-task-2)
+      - [Using Docker Compose](#using-docker-compose-1)
+****
+
+## Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+- [Docker](https://docs.docker.com/get-docker/) (newer versions already include Docker Compose)
+  - [Docker Compose](https://docs.docker.com/compose/install/) (only needed for older Docker versions)
+- [Task](https://taskfile.dev/installation/) (optional but recommended)
+
+> [!NOTE]
+> The `Taskfile` will automatically detect your Docker version and use the appropriate command (`docker-compose` or `docker compose`).
+
+## Setup Steps
+
+### 1. Clone the Repository
+
+```bash
+git clone https://github.com/TheRestartProject/restarters.net.git
+cd restarters.net
+```
+
+### 2. Environment Configuration
+
+Copy the example environment file:
+
+```bash
+cp .env.example .env
+```
+
+You may need to adjust the following settings in your `.env` file:
+- Set `APP_URL` to `http://localhost:8001` (matching the port in docker-compose.yml)
```

Comment:
Any reason not to change the version in .env.example?  

It's used in `travis.yml`, but we can delete that file - we use CircleCI now.

It's also used in `docker_run.sh`

Reply:
>Any reason not to change the version in .env.example?

Are you referring to changing `APP_URL` to the following?
```diff
- APP_URL=http://restarters.test:8000
+ APP_URL=http://localhost:8001
```

If so, then I didn't change it because I lack knowledge.

>It's used in travis.yml, but we can delete that file - we use CircleCI now.
>It's also used in docker_run.sh

As you mentioned, the `.env` file, and thus `APP_URL`, is used in both these files. However, I'm not sure why `APP_URL` was set to that value in the first place as the original commit did not have much information.

https://github.com/TheRestartProject/restarters.net/commit/47afb56355130734cc9bfe976db86e00bde43c3a

I opted to leave it alone since I lacked insight.

Reply:
Change it, please, and delete `travis.yml`.  We'll deal with any issues that creates on our side.

Reply:
Addressed in https://github.com/TheRestartProject/restarters.net/pull/795/commits/a9a619cd6f03243766617be3dc44ba56ce6d00d9